### PR TITLE
refactor: Rework Proxy Setup to use KongRoute struct for configuration

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?
Proxy Setup uses repurposed ClientInfo struct for the Route configuration which has become an issue once the standards keys changed to service keys. This is because the key name is also used for the route name

## What is the new behavior?
Now Proxy Setup uses existing KongRoute struct that has extra name property for the route name. The Route configuration is properly named `Routes`, no longer `Clients`

## Does this PR introduce a breaking change?

- [x ] Yes
- [ ] No

BREAKING CHANGE: Names for Route configuration has changed

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: cmd/security-proxy-setup/res/configuration.toml [hunk 1]
```diff
@@ -54,38 +54,45 @@ RetryWaitPeriod = "1s"
   [SecretStore.Authentication]
   AuthType = "X-Vault-Token"
 
-[Clients]
-  [Clients.CoreData]
+[Routes]
+  [Routes.CoreData]
+  Name = "coredata"
```
Comment (jim-wang-intel) [inline cmd/security-proxy-setup/res/configuration.toml:59]: could you add a comment on the name is always being lower cases as internally we setup kong route that way?

**Review by jim-wang-intel — COMMENTED** [review]
overall LGTM

File: cmd/security-proxy-setup/res/configuration.toml (hunk 1 above)
Comment (lenny-intel) [reply cmd/security-proxy-setup/res/configuration.toml:59]: Code is still forcing it to lowercase, so case here doesn't matter.

**Review by bnevis-i — APPROVED** [review]

File: cmd/security-proxy-setup/res/configuration.toml (hunk 1 above)
Comment (jim-wang-intel) [reply cmd/security-proxy-setup/res/configuration.toml:59]: i think it matters. the service name is the key to the map, so the coredata != CoreData in terms of map's key.

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
map key is case sensitive

File: cmd/security-proxy-setup/res/configuration.toml (hunk 1 above)
Comment (lenny-intel) [reply cmd/security-proxy-setup/res/configuration.toml:59]: Lowercased `name` is used to create the proxy on line 149 & 153 of service.go

Comment (jim-wang-intel) [reply cmd/security-proxy-setup/res/configuration.toml:59]: ok, i am confused by line 254-256, which is using the name directly from config.route. so if one puts mix or big cases in the name of config file then the name to the map will be like that.
```go
	for serviceName, route := range s.configuration.Routes {
		merged[serviceName] = route
	}
```

**Review by jim-wang-intel — APPROVED** [review]

File: cmd/security-proxy-setup/res/configuration.toml (hunk 1 above)
Comment (lenny-intel) [reply cmd/security-proxy-setup/res/configuration.toml:59]: `serviceName` is just the map key which is not used for the route. `route` now has the Name filed which is used on lines 149 & 153 to create the actual proxy route. We no longer use the map key as part of the `proxy route`.

Comment (jim-wang-intel) [reply cmd/security-proxy-setup/res/configuration.toml:59]: i got that, it's just that we have this new "name" field from config now and it creates the contention in the map's key with it. admitted it's the same behavior as before in the map, so seeing resolved.

**Pull request merged**